### PR TITLE
Prometheus Operator installs kube-state-metrics

### LINKED ISSUE #1408: Improve (!?) documentation for enabling VerticalPodAutoscaler
**What happened**:

It took me more time than I would have liked to understand how to enable `VerticalPodAutoscaler` metrics.

The documentation [hints](https://github.com/kubernetes/kube-state-metrics#enabling-verticalpodautoscalers) at the solution but would benefit from an additional explanation for running `kube-state-metrics` with `--resource=${RESOURCES},verticalpodautoscalers`.

**What you expected to happen**:

A more detailed explanation of what's needed and how to enable this with e.g. Prometheus Operator.

Perhaps back-filling issues with VPA and the Prometheus Operator to provide guidance on this too.

It's very helpful just confusing to resolve.

**How to reproduce it (as minimally and precisely as possible)**:

I'm running MicroK8s 1.20 and `microk8s enable prometheus` and have VPA installed:

```bash
kubectl get crd/verticalpodautoscalers.autoscaling.k8s.io \
--output=jsonpath="{.spec.versions[*].name}"
v1beta1 v1beta2 v1
```
After cloning this repo and deploying `./examples/standard`, I realized that the Prometheus Operator bundles `kube-state-metrics` but then realized that it does not enable `verticalpodautoscalers` (by default).

```bash
kubectl logs deployment/kube-state-metrics \
--namespace=monitoring \
--container=kube-state-metrics

...
Active resources:
  certificatesigningrequests,configmaps,cronjobs,
  daemonsets,deployments,endpoints,horizontalpodautoscalers,
  ingresses,jobs,leases,limitranges,mutatingwebhookconfigurations,
  namespaces,networkpolicies,nodes,persistentvolumeclaims,
  persistentvolumes,poddisruptionbudgets,pods,replicasets,
  replicationcontrollers,resourcequotas,secrets,services,
  statefulsets,storageclasses,validatingwebhookconfigurations,
  volumeattachments
...
```
and:

```bash
kubectl get deployment/kube-state-metrics \
--namespace=monitoring \
--output=jsonpath="{.spec.template.spec.containers[?(@.name=='kube-state-metrics')]}" \
| jq .
```

Yields:

```JSON
{
  "args": [
    "--host=127.0.0.1",
    "--port=8081",
    "--telemetry-host=127.0.0.1",
    "--telemetry-port=8082",
  ],
  "image": ".../kube-state-metrics:v1.9.7",
  "imagePullPolicy": "IfNotPresent",
  "name": "kube-state-metrics",
  "resources": {},
  "terminationMessagePath": "/dev/termination-log",
  "terminationMessagePolicy": "File"
}
```

I don't know how to configure the metrics correctlypermanently but I managed to tweak it:

```bash
RESOURCES="certificatesigningrequests,configmaps,cronjobs,"\
"daemonsets,deployments,endpoints,horizontalpodautoscalers,"\
"ingresses,jobs,leases,limitranges,mutatingwebhookconfigurations,"\
"namespaces,networkpolicies,nodes,persistentvolumeclaims,"\
"persistentvolumes,poddisruptionbudgets,pods,replicasets,"\
"replicationcontrollers,resourcequotas,secrets,services,"\
"statefulsets,storageclasses,validatingwebhookconfigurations,"\
"verticalpodautoscalers,volumeattachments"

PATCH="[{"\
"'op': 'add',"\
"'path':'/spec/template/spec/containers/0/args/-',"\
"'value':\"--resources=${RESOURCES}\"}]"


kubectl patch deployment/kube-state-metrics \
--namespace=monitoring \
--type=json \
--patch="${PATCH}" 
```

**Anything else we need to know?**:

Hopefully others who struggle with this will find this issue.

I blogged about the experience [here](https://pretired.dazwilkin.com/posts/210305/)

**Environment**:
- kube-state-metrics version: 
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Other info:

### PR DESCRIPTION
**Pull request by DazWilkin**
**What this PR does / why we need it**:

The Prometheus Operator ([`kube-prometheus`](https://github.com/prometheus-operator/kube-prometheus)) includes `kube-state-metrics`. There is no need to install `kube-state-metrics` if you're going to use the Prometheus Operator.

**Which issue(s) this PR fixes**:
Fixes #1408 

<strike>I'm going to file an issue against Prometheus Operator to determine whether its Deployment can optionally include the specification of kube-state-metrics Resources (e.g. VPA).</strike>

### CODE REVIEW

File: README.md
Comment (lilic) [inline README.md:257]: Prometheus operator does not install kube-state-metrics, but the kube-promethus stack does, so this is a bit misleading.

File: README.md
Comment (lilic) [inline README.md:269]: Master should not reference 1.9.8 images.

**Review by lilic — COMMENTED** [review]
Thanks for this, as I am the maintainer of Prometheus operator and kube-prometheus projects, I cannot judge if this is a good thing to include in this repo. As those projects are not part of the k8s org.

@tariq1890 your call on this one!

File: README.md
Comment (DazWilkin) [reply README.md:257]: Is the vernacular that a stack include an Operator?

As someone outside of the core this isn't a differentiation with which I'm familiar.

If I change this to say "The kube-prometheus stack (including the Prometheus Operator) installs..."?

File: README.md
Comment (DazWilkin) [reply README.md:269]: If I change this to say `image: k8s.gcr.io/kube-state-metrics/kube-state-metrics` would that be OK?

File: README.md
Comment (lilic) [reply README.md:257]: the prometheus operator does not install kube-state-metrics and knows nothing about kube-state-metrics itself. kube-prometheus has the manifests and knowledge around it, coming from this projects jsonnet mixin. So mentioning Prometheus operator is misleading unless you are referring to the org? We tend to refer to it as kube-prometheus stack.

File: README.md
Comment (lilic) [reply README.md:269]: Not sure this step is necessary as people have a tendency of copy pasteing yaml around so I would prefer if they did it from working examples. If that makes sense? :) 

Also we autogenerate all our code, so adding things like this would mean manual updating, we can embed the code instead if you think its important to reference it here?

File: README.md
Comment (DazWilkin) [reply README.md]: Fixed: https://github.com/kubernetes/kube-state-metrics/pull/1412/commits/1b44f89f78b73d9b211aa6c69147642e78bca71a

File: README.md
Comment (DazWilkin) [reply README.md:269]: kube-prometheus runs kube-state-metrics but it does so without specifying `--resources`.

As a result, only the default resources are enabled.

If the users wishes to enable non-default metrics (e.g. VPA), they must include the `--resources` flag, list all the default resources and include the non-default metric(s).

Perhaps replace the text with some variant of that explanation?

**Comment by tariq1890** [comment]
@lilic Adding this is okay by me 👍

File: README.md
Comment (tariq1890) [inline README.md:259]:
```suggestion
The `kube-state-metrics` Deployment [manifest] overrides several command line arguments of the kube-state-metrics binary:
```

File: README.md
Comment (tariq1890) [inline README.md:255]:
```suggestion
### For the Users of Prometheus Operator
```

File: README.md
Comment (lilic) [reply README.md:269]: Yes, that is out of the box, but you can add a jsonnet patch for this if you are using kube-prometheus we have that documented in there -> https://github.com/prometheus-operator/kube-prometheus#customizing-kube-prometheus

I would suggest to revise this to "Out of the box kube-promethus ships with default mixin configuration if you want to customize have a look at <https://github.com/prometheus-operator/kube-prometheus#customizing-kube-prometheus>" What do you think?

File: README.md
Comment (lilic) [reply README.md:255]: This should be maybe: 

For the users of prometheus-operator/kube-prometheus stack?

Comment (tariq1890) [reply README.md:255]: Sounds good!

File: README.md
Comment (DazWilkin) [reply README.md]: Fixed: https://github.com/kubernetes/kube-state-metrics/pull/1412/commits/86689d45db2730aff799e495729f29993734087c

File: README.md
Comment (DazWilkin) [reply README.md]: This text has been remove (https://github.com/kubernetes/kube-state-metrics/pull/1412/commits/86689d45db2730aff799e495729f29993734087c)

File: README.md
Comment (DazWilkin) [reply README.md]: Fixed: https://github.com/kubernetes/kube-state-metrics/pull/1412/commits/c27da11e3ba5c11a9c0e0bed0cd53608bd7cb5aa

**Review by lilic — COMMENTED** [review]
Very nice, do you mind just squashing your commits as I am not sure the bot that merges is smart enough to squash and merge, otherwise looks great to me! Thank you!

**Comment by DazWilkin** [comment]
Done.

**Review by lilic — APPROVED** [review]
/lgtm

Thanks!

**Pull request merged**
